tests/integration_tests/test_version_command.py

```diff
@@ -32,3 +33,44 @@ def or_fail():
 
     await helpers.get_test_channel().send("!version")
     await helpers.wait_for(version_to_appear, or_fail)
+
+
+@helpers.threaded_async
+async def test_returns_error_on_missing_role():
+    """
+    This function makes sure that members who do not have the "Defender" role can't use the !version command.
+    The function removes the "Defender" role from DofDevBot, then it makes the bot use the command !version as a user.
+    The bot shouldn't be able to have the version as result for his command due to his role not being the one required.
+    At the end the "Defender" role is added back to DofDevBot.
+    """
+    async def error_to_appear():
+        """
+        Calling the command without a Defender role should result in the error message being printed.
+        """
+        message = helpers.get_test_channel().last_message
+        if message and message.author.name == "DofDevBotApplication" :
+            assert message.embeds[0].title == "Role 'Defender' is required to run this command."
+            return True
+        Log.debug("Waiting for the response to the !version command to appear")
+
+
+    def or_fail():
+        """
+        Otherwise, fail the test.
+        """
+        pytest.fail("Timed out waiting for response to the !version command")
+
+    def get_tester_bot_member() -> discord.Member:
+        """
+        Helper command to retrieve the testing bot using the unique identifier.
+        """
+        for member in helpers.dof_bot.get_all_members():
+            if member.id == 720954603046305855:
+                return member
+
+    tester_bot_member = get_tester_bot_member()
+    defender_role = discord.utils.get(helpers.dof_bot.guild.roles, name="Defender")
+    await tester_bot_member.remove_roles(defender_role)
+    await helpers.get_test_channel().send("!version")
+    await helpers.wait_for(error_to_appear, or_fail)
+    await tester_bot_member.add_roles(defender_role)
```

Comment:
Add spacing and comments explaining some unclear lines, please. E.g. first line should be commented that we need to have a `member` object to be able to call `remove_roles` on the bot, 2nd and 3rd lines should be together and explained with a comment that a role must be removed before the test happens, the last line should have a comment too etc.